misc/python/materialize/cli/mzconduct.py

```diff
@@ -806,251 +807,190 @@ def run(self, comp: Composition, workflow: Workflow) -> None:
             say(f"INFO: error purging topics: {e}")
 
 
-class ChaosDockerWorkflowStep(WorkflowStep):
-    """Stops/pauses and starts the designated Docker service.
-
-    Params:
-        service: target Docker service, will be used to grep for container id
-                 NOTE: service name must be unique to correctly match the container id
-
-        other_service: if provided, run only as long as this service is running
-        duration: if provided, run for this duration
-
-        run_cmd: command to start the Docker container
-        run_time: seconds to spend running each loop
-        stop_cmd: command to stop the Docker container
-        stopped_time: seconds to spend stopped each loop
+@Steps.register("random-chaos")
+class RandomChaos(WorkflowStep):
     """
+    Add random chaos to running Docker containers.
+
+    :param chaos:          List containing types of chaos to add. If not provided,
+                           will default to 'default_chaos'.
+    :param services:       List of target Docker services for chaos. If not provided,
+                           will default to all running Docker services.
+    :param other_service:  Chaos will be randomly added to Docker services as long as
+                           'other_service' is running. If not provided, chaos will be
+                           added forever.
+    """
+
+    default_chaos = [
+        "pause",
+        "stop",
+        "kill",
+        "delay",
+        "rate",
+        "loss",
+        "duplicate",
+        "corrupt",
+    ]
 
     def __init__(
-        self,
-        service: str,
-        other_service: str,
-        duration: int,
-        run_cmd: str,
-        stop_cmd: str,
-        run_time: int,
-        stop_time: int,
+        self, chaos: List[str] = [], services: List[str] = [], other_service: str = "",
```

Comment:
To prevent refactoring bugs these parameters should default to `None` instead of the empty list.

Reply:
Our `bin/pre-push` lint won't allow me to put a `None` default for a `List` type:
```
error: Incompatible default for argument "services" (default has type "None", argument has type "List[str]")
```